framework/src/media/PlayerObserverWorker.cpp

```diff
@@ -38,44 +38,12 @@ PlayerObserverWorker& PlayerObserverWorker::getWorker()
 int PlayerObserverWorker::entry()
 {
 	while (mIsRunning) {
-		std::function<void()> run = mObserverQueue.deQueue();
+		std::function<void()> run = mWorkerQueue.deQueue();
```

Comment:
Let's optimize entry more.
Actually entry's role is below
```
while (running) {
      do_something();
      dequeue();
}
```
You can make abstract function do_something, and inherited class can implement do_something.
Also entry is the only place access mWorkerQueue directly, I think this can be hided if we give  mWorkerQueue as a parameter when it binded.

Reply:
I agree with your idea, but I think the implementation takes more time. Could I summit in another PR?

Reply:
Of course.